openvasreporting/libs/parsed_data.py

```diff
@@ -31,32 +34,45 @@ def __init__(self, number, protocol="tcp"):
         if not isinstance(protocol, str):
             raise TypeError("Expected basestring, got '{}' instead".format(type(protocol)))
 
+        if not isinstance(result, str):
+            raise TypeError("Expected basestring, got '{}' instead".format(type(result)))
+
         self.number = number
         self.protocol = protocol
+        self.result = result
 
+    # Modified to include result in structure
     @staticmethod
-    def string2port(info):
+    def string2port(info,result):
         """
         Extract port number, protocol and description from an string.
+	resturn a port class with seperate port, protocol and result
 
         ..note:
             Raises value error if information can't be processed.
 
-        # >>> p=Port.string2port("2000/tcp")
+        # >>> p=Port.string2port("2000/tcp","result string")
         # >>> print p.number
           2000
         # >>> print p.proto
           "tcp"
+	# >>> print p.result
+	  "string"
 
-        # >>> p=Port.string2port("general/icmp")
+        # >>> p=Port.string2port("general/icmp", "string test")
         # >>> print p.number
           0
         # >>> print p.proto
           "icmp"
+	# >>> print p.result
+	  "stringntest"
```

Comment:
Looks like this output doesn't match your example input

Shouldn't this be `string test`